api/apps/api/src/modules/scenarios/locks/lock.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Connection, Repository } from 'typeorm';
+
+import { ScenarioLockEntity } from '@marxan-api/modules/scenarios/locks/scenario.lock.entity';
+import { LockedScenarioError } from './errors/locked.scenario.error';
+import { LockNotFoundError } from './errors/lock.not.found.error';
+
+@Injectable()
+export class LockService {
+  constructor(
+    @InjectRepository(ScenarioLockEntity)
+    private readonly locksRepo: Repository<ScenarioLockEntity>,
+    private connection: Connection,
+  ) {}
+
+  async acquireLock(scenarioId: string, userId: string): Promise<void> {
+    return this.connection.transaction(async (entityManager) => {
+      const existingLock = await entityManager.find(ScenarioLockEntity, {
+        where: { scenarioId, userId },
+      });
+
+      if (existingLock.length > 0) {
+        throw new LockedScenarioError();
```

Comment:
I think it would be nice to keep the convention of returning `Either` with possible errors (to at lest let consumer know what can go wrong within intention). 
Also Error instances themselves are bit trickier to handle for consumer.

Reply:
@kgajowy https://github.com/Vizzuality/marxan-cloud/pull/673/commits/f8f415313c7d05aa8793f34fc7107515a4958e76